libraries/app/database_api.cpp

```diff
@@ -1554,11 +1554,17 @@ vector<market_trade> database_api_impl::get_trade_history( const string& base,
          {
             trade.sequence = -itr->key.sequence;
             trade.side1_account_id = itr->op.account_id;
-            trade.side1_was_selling = itr->op.pays.asset_id == assets[0]->id ? base : quote;
+            if(itr->op.receives.asset_id == assets[0]->id)
+               trade.type = "buy";
+            else
+               trade.type = "sell";
          }
          else {
             trade.side2_account_id = itr->op.account_id;
-            trade.side2_was_selling = itr->op.pays.asset_id == assets[0]->id ? base : quote;
+            if(itr->op.receives.asset_id == assets[1]->id)
+               trade.type = "sell";
+            else
+               trade.type = "buy";
```

Comment:
We can remove this code block, since there will always be a maker, the `type` data will be filled in the `if` block. (Actually there is not always a taker, e.g. in the scenario of a global settlement).